crates/node/p2p/src/gossip/builder.rs

```diff
@@ -124,7 +124,7 @@ impl GossipDriverBuilder {
     pub fn build(mut self) -> Result<GossipDriver, GossipDriverBuilderError> {
         // Extract builder arguments
         let timeout = self.timeout.take().unwrap_or(Duration::from_secs(60));
-        let keypair = self.keypair.take().unwrap_or(Keypair::generate_secp256k1());
+        let keypair = self.keypair.take().ok_or(GossipDriverBuilderError::MissingKeyPair)?;
```

Comment:
Nice, makes a lot more sense to leave it up to the user to set sensible defaults